src/merkle/smt/full/concurrent/tests.rs

```diff
@@ -479,6 +482,67 @@ fn test_smt_construction_with_entries_unsorted() {
     assert_eq!(smt, control);
 }
 
+#[test]
+fn test_smt_construction_with_entries_duplicate_keys() {
+    let entries = [
+        (RpoDigest::new([ONE, ONE, ONE, Felt::new(16)]), [ONE; 4]),
+        (RpoDigest::new([ONE; 4]), [ONE; 4]),
+        (RpoDigest::new([ONE, ONE, ONE, Felt::new(16)]), [ONE; 4]),
+    ];
+    let result = Smt::with_entries(entries);
+    assert!(matches!(result, Err(MerkleError::DuplicateValuesForIndex(_))));
+
+    if let Err(MerkleError::DuplicateValuesForIndex(col)) = result {
+        let expected_col = Smt::key_to_leaf_index(&entries[0].0).index.value();
+        assert_eq!(col, expected_col);
+    }
```

Comment:
```suggestion
    let expected_col = Smt::key_to_leaf_index(&entries[0].0).index.value();
    let err = Smt::with_entries(entries).unwrap_err();
    assert_matches!(err, MerkleError::DuplicateValuesForIndex(col) if col == expected_col);
```

Nit/Suggestion: This could be written a bit more concisely.